test/testsuites/cscpackagetests.go

```diff
@@ -0,0 +1,125 @@
+package testsuites
+
+import (
+	"fmt"
+	"testing"
+
+	olm "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	operator "github.com/operator-framework/operator-marketplace/pkg/apis/operators/v1"
+	"github.com/operator-framework/operator-marketplace/test/helpers"
+	"github.com/operator-framework/operator-sdk/pkg/test"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apps "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// CscPackages is a test suit that confirms that the packages field within
+// a CatalogSourceConfig is handled correctly.
+func CscPackages(t *testing.T) {
+	t.Run("non-existing-package", testCscWithNonExistingPackage)
+}
+
+// testCscWithNonExistingPackageName creates context and calls the
+// cscWithNonExistingPackageName test case.
+func testCscWithNonExistingPackage(t *testing.T) {
+	ctx := test.NewTestCtx(t)
+	defer ctx.Cleanup()
+
+	// Get global framework variables.
+	f := test.Global
+
+	// Run tests.
+	if err := cscWithNonExistingPackage(t, f, ctx); err != nil {
+		t.Fatal(err)
+	}
+}
+
+// cscWithNonExistingPackage is a test case that creates a CatalogSourceConfig
+// with a package that doesn't exist and ensures that child resources are not 
+// created.
+func cscWithNonExistingPackage(t *testing.T, f *test.Framework, ctx *test.TestCtx) error {
+	// Get test namespace
+	namespace, err := ctx.GetNamespace()
+	if err != nil {
+		return fmt.Errorf("Could not get namespace: %v", err)
+	}
+
+	// nonExistingPackageCSCName is the name of the catalogsourceconfig that points
+	// to a non-existing Package
+	nonExistingPackageCSCName := "non-existing-package-operators"
+
+	// packageName is the non-existing package
+	packageName := "non-existing-package"
+
+	// Create a new catalogsourceconfig with a non-existing Package
+	nonExistingPackageCSC := &operator.CatalogSourceConfig{
+		TypeMeta: metav1.TypeMeta{
+			Kind: operator.CatalogSourceConfigKind,
+		}, ObjectMeta: metav1.ObjectMeta{
+			Name:      nonExistingPackageCSCName,
+			Namespace: namespace,
+		},
+		Spec: operator.CatalogSourceConfigSpec{
+			TargetNamespace: namespace,
+			Packages:        packageName,
+		}}
+	err = helpers.CreateRuntimeObject(f, ctx, nonExistingPackageCSC)
+	if err != nil {
+		return err
+	}
+
+	// Check that we created the catalogsourceconfig with a non-existing package
+	resultCatalogSourceConfig := &operator.CatalogSourceConfig{}
+	err = helpers.WaitForResult(t, resultCatalogSourceConfig, namespace, nonExistingPackageCSCName)
```

Comment:
You run this same WaitForResult function below when checking for the phase, you can remove this.